### ISSUE TITLE
UI Folders and Requests list dragging and movement issues

### ISSUE THREAD

**Issue by DarceyLloyd** [getinsomnia/insomnia#5831]
### Expected Behavior

Drag and drop doesn't work, well not fully.

**Issue 1 - folder movement:**
1. Make a few folders and make a few requests in each folder (no items outside of folders)
2. Create new folder (put it at the top)
3. Try and drag that folder in-between other folders (I've not been able to do this), it always ends up in another folder.
4. After doing this there is no way to move that folder back to the top at root level
5. There is no way to move that folder out of another folder, but there is a hack to get it to work. You create 2 request outside of the folders and then drag that folder under them or in between them.

Desired functionality
You should be able to drag and drop to anywhere in the order you want whether it be at root or nested levels.

**Issue 2 - request / entry movement:**
1. Make a few folders and make a few requests in each folder (no items outside of folders)
2. Try and drag 1 of the requests in 1 of the folders to the root level, this works first time but not at the desired location of drag indicator and then you can no longer drag an item from a request in a folder to root level at all anywhere unless you create 2 new requests at root to just get the drag and drop to work out location better but then you will never be able to place them in the root at a position of wanting.
3. There is no way to drag that request to the top at root level.

You should be able to drag and drop to anywhere in the order you want whether it be at root or nested levels.

This has been an issue for as long as I've been using Insomnia.

**Suggestion**
To see how it should work and a suggestion on key control, download KeePass (security password manager, it's free), make folders and some sub folders and make some entries in each. You can use short cut keys (alt and up and down) to move items up and down and also drag items around folder in folder and entries in folder to anywhere you want.

- ALT + UP move item up
- ALT + UP move item up (if at top of folder, will move out to root above)
- ALT + UP move item up (if at root and folder above it will move above it)
- ALT + UP move item up (if at root top, do nothing)
- ALT + DOWN move item down
- ALT + DOWN move item down (if at bottom of folder, will move out to root below)
- ALT + DOWN move item down (if at root bottom, do nothing)

Mouse drag should be able to handle all this also.

### Actual Behavior

See above...

### Reproduction Steps

See above...

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Additional Information

This has been an issue for as long as I've been using Insomnia. I've always had to find hacks to try and resolve the issue. 

Exporting the json and re-importing after manually editing it is sometimes the only way (which I just can't be bothered to do anymore)

### Insomnia Version

2023.1.0

### What operating system are you using?

Windows

### Operating System Version

Windows 10 Pro 22H2 - 10.0.19045 Buid 19045

### Installation method

Download from net, run exe.

**Comment by jackkav** [comment]
fixed in #6323

**Comment by ozmium** [comment]
This bug is still happening for me in the most recent Insomnia version:

```
Version: Insomnia 11.0.1
Build date: 27/03/2025
OS: Linux x64 6.8.0-52-generic
Electron: 35.0.2
Node: 22.14.0
Node ABI: 133
V8: 13.4.114.19-electron.0
Architecture: x64
```

It's a bad bug that has been happening for several years.

I found a new way to workaround it, but it takes a lot of patience. See this video:

https://github.com/user-attachments/assets/dfda3ae9-ebb7-455d-b840-324eb2526436

The workaround steps are:

1. In the Sidebar on the left, create a New Folder as normal. Unfortunately, Insomnia will add it as a nested folder, which is not what the user wants.
2. Locate an existing top-level folder in your list of endpoints (which could be the current folder, if you want)
3. Move/drag your New Folder to one item above that top-level folder. The New Folder will now be moved to the root level, or the parent level, which is what the user intended.
4. If you want to move the New Folder down again, you have to repeat the process:
5. So drag the New Folder downwards. Insomnia will move it into a nested folder, unhelpfully.
6. Then drag the New Folder to 1 level above the top-level folder. It will now be at the root.
